contents/julia_101.md

```diff
@@ -423,55 +517,2108 @@ logarithm(10)
+
+To begin, let's import the `Dates` module:
+
+```julia
+using Dates
+```
+
+#### `Date` and `DateTime` Types {#sec:dates_types}
+
+The `Dates` standard library module has **two types for working with dates**:
+
+1. `Date`: representing time in days; and
+2. `DateTime`: representing time in millisecond precision.
+
+We can construct `Date` and `DateTime` with the default constructor either by specifying an integer to represent year, month, day, hours and so on:
+
+```jl
+sco(
+"""
+Date(1987) # year
+"""
+)
+```
+
+```jl
+sco(
+"""
+Date(1987, 9) # month
+"""
+)
+```
+
+```jl
+sco(
+"""
+Date(1987, 9, 13) # day
+"""
+)
+```
+
+```jl
+sco(
+"""
+DateTime(1987, 9, 13, 21) # hour
+"""
+)
+```
+
+```jl
+sco(
+"""
+DateTime(1987, 9, 13, 21, 21) # minute
+"""
+)
+```
+
+For the curious, September 13th 1987, 21:21 is the official time of birth of the first author, Jose.
+
+
+We can also pass `Period` types to the default constructor.
+**`Period` types are the human-equivalent representation of time** for the computer.
+Julia's `Dates` have the following `Period` abstract subtypes:
+
+```jl
+sco(
+"""
+subtypes(Period)
+"""
+)
+```
+
+Which divide into the following concrete types, they are pretty much self-explanatory:
+
+```jl
+sco(
+"""
+subtypes(DatePeriod)
+"""
+)
+```
+
+```jl
+sco(
+"""
+subtypes(TimePeriod)
+"""
+)
+```
+
+So we could alternatively construct Jose's official time of birth as:
+
+```jl
+sco(
+"""
+DateTime(Year(1987), Month(9), Day(13), Hour(21), Minute(21))
+"""
+)
+```
+
+#### Parsing Dates {#sec:dates_parsing}
+
+Most of the time we won't be constructing `Date` or `DateTime` instances from scratch.
+Actually, we probably will be **parsing strings as `Date` or `DateTime` types**.
+
+The `Date` and `DateTime` constructors can be fed a string and a format string.
+For example, the string `"20210101"` representing January 1st can be parsed with:
+
+```jl
+sco(
+"""
+Date("19870913", "yyyymmdd")
+"""
+)
+```
+
+Notice that the second argument is a string representation of the format.
+We have the first four digits representing year `y`, followed by two digits for month `m` and finally two digits for day `d`.
+
+It also works for timestamps with `DateTime`:
+
+```jl
+sco(
+"""
+DateTime("1987-09-13T21:21:00", "yyyy-mm-ddTHH:MM:SS")
+"""
+)
+```
+
+You can find more on how to specify different format as strings in the [Julia `Dates`' documentation](https://docs.julialang.org/en/v1/stdlib/Dates/#Dates.DateFormat).
+Don't worry if you have to revisit it all the time, we ourselves have to do it all the time when working with dates and timestamps.
+
+According to [Julia `Dates`' documentation](https://docs.julialang.org/en/v1/stdlib/Dates/#Constructors), using the `Date(date_string,format_string)` function is fine if only called a few times.
+If there are many similarly formatted date strings to parse however, it is much more efficient to first create a `DateFormat` type, and then pass it instead of a raw format string.
```

Comment:
Would this be a real issue to our users? They can read about it in the documentation too

Reply:
I decided to cover since we are using it in DataFrames as timeseries analysis.